include/exadg/grid/grid.cpp

```diff
@@ -160,10 +164,52 @@ Grid<dim>::create_triangulation(
     {
       // we need to generate the coarse triangulations explicitly
 
-      // TODO
-      // construct all the coarser triangulations
+      // resize the empty coarse triangulations vector
+      coarse_triangulations = std::vector<std::shared_ptr<dealii::Triangulation<dim> const>>(
+        triangulation->n_global_levels());
+
+      // we start with the finest level
+      unsigned int              level        = triangulation->n_global_levels() - 1;
+      std::vector<unsigned int> refine_local = vector_local_refinements;
```

Comment:
Why do you define these variables here? Wouldn't it make more sense to shift these lines to line 187 below.

Reply:
done!

Reply:
I also moved lines 167-169.